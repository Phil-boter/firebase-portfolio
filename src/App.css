:root {
    --var-color_one: #0ff;
    --var-color_two: #030303;
    --var-color_three: #ffffee;
    --var-color_gray: rgb(247, 247, 247);
    --var-color_footer: rgb(240, 240, 240);
    --var-color_green: #00ff96;
    --var-color_white: white;
    --var-color_black: black;
    --var-color_projectHeader: rgb(218, 38, 205);
    --var-color_headline: yellow;
}

* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: Montserrat;
    list-style: none;
    font-size: 16px;
}
