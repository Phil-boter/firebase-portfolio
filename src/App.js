import "./App.css";
import { DataProvider } from "./context/DataProvider";
import { LanguageContext } from "./context/LanguageProvider";
import React, { useState } from "react";
import { BrowserRouter, Router, Route, Switch } from "react-router-dom";

// -------Inintializing fireBase--------------
import firebase from "./Firebase/config";
// -------Inintializing fireBase--------------

// _______importing Routes ___________________
import Projects from "./routes/projects/Index";
import Main from "./routes/main/Index";
import Contact from "./routes/contact/Index";
import About from "./routes/about/Index";
import SingleProject from "./routes/singleProject/Index";

// _______importing Routes ___________________

import Navigation from "./components/navigation/Index";
import ScrollToTopButton from "./components/scrollToTopButton/Index";
import Footer from "./components/footer/Index";

function App() {
    console.log("firebase", firebase);

    const [language, setLanguage] = useState("en");

    function toggleLanguage() {
        setLanguage((language) => (language === "en" ? "de" : "en"));
    }
    return (
        <BrowserRouter>
            <Switch>
                <LanguageContext.Provider
                    value={{ language, setLanguage, toggleLanguage }}
                >
                    <DataProvider>
                        <Navigation />

                        <Route exact path="/" render={() => <Main />} />
                        <Route path="/projects" render={() => <Projects />} />
                        <Route path="/contact" render={() => <Contact />} />
                        <Route path="/about" render={() => <About />} />
                        <Route
                            path="/singleProject/:id"
                            render={(props) => (
                                <SingleProject
                                    projectId={props.match.params.id}
                                />
                            )}
                        />

                        <Footer />
                        <ScrollToTopButton />
                    </DataProvider>
                </LanguageContext.Provider>
            </Switch>
        </BrowserRouter>
    );
}

export default App;
