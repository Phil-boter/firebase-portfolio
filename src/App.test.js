import React from "react";

import { render, mount } from "enzyme";
import App from "./App";

describe("App", () => {
    let wrapper = render(<App />);
    it("should render App component correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
    wrapper = mount(<App />).find(".app-test");
    expect(wrapper.hasClass("app-test")).toEqual(true);
});
