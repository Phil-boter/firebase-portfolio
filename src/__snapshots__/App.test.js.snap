// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App component correctly 1`] = `
<div
  className="app-test"
/>
`;
